fastlane_core/spec/update_checker_spec.rb

```diff
@@ -35,5 +35,22 @@
         expect(FastlaneCore::UpdateChecker.update_available?(name, '0.9.1.pre1')).to eq(true)
       end
     end
+
+    describe "#p_hash?" do
+      it "chooses the correct param for package name for supply" do
+        args = ["--skip_upload_screenshots", "-a", "beta", "-p", "com.test.app"]
+        expect(FastlaneCore::UpdateChecker.p_hash(args, 'supply')).to eq("2d195f58610c784cd135f22bf54561487e163fb25a10726bf0a7349d645a052e")
+      end
+
+      it "chooses the correct param for package name for screengrab" do
+        args = ["--skip_open_summary", "-a", "com.test.app", "-p", "com.test.app.test"]
+        expect(FastlaneCore::UpdateChecker.p_hash(args, 'screengrab')).to eq("2d195f58610c784cd135f22bf54561487e163fb25a10726bf0a7349d645a052e")
+      end
+
+      it "chooses the correct param for package name for gym" do
+        args = ["--clean", "-a", "com.test.app", "-p", "test.xcodeproj"]
+        expect(FastlaneCore::UpdateChecker.p_hash(args, 'gym')).to eq("181a84115f14300fefd82373db6da188af898f247c67b2cd9ff713f4cf7f69e6")
```

Comment:
I like that we're testing this end-to-end, but these hash values feel pretty opaque to me as indicators of success.

Can we define some helpers to help make the relationship between the inputs and expected outputs clearer? Something like:

```ruby
def android_hash_of(value)
  hash_of("android_project_#{value}")
end

def hash_of(value)
  Digest::SHA256.hexdigest("p#{value}fastlan3_SAlt")
end

# ...

expect(FastlaneCore::UpdateChecker.p_hash(args, 'screengrab')).to eq(android_hash_of("com.test.app"))

#...

expect(FastlaneCore::UpdateChecker.p_hash(args, 'gym')).to eq(hash_of("com.test.app"))
```